va/va_fei_h264.h

```diff
@@ -0,0 +1,396 @@
+/*
+ * Copyright (c) 2007-2017 Intel Corporation. All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sub license, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice (including the
+ * next paragraph) shall be included in all copies or substantial portions
+ * of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+ * OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT.
+ * IN NO EVENT SHALL INTEL AND/OR ITS SUPPLIERS BE LIABLE FOR
+ * ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * \file va_fei_h264.h
+ * \brief The FEI encoding H264 special API
+ */
+
+#ifndef VA_FEI_H264_H
+#define VA_FEI_H264_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
+#include "va_fei.h"
+
+/** \brief FEI frame level control buffer for H.264 */
+typedef struct _VAEncMiscParameterFEIFrameControlH264
+{
+    uint32_t      function; /* one of the VAConfigAttribFEIFunctionType values */
+    /** \brief MB (16x16) control input buffer. It is valid only when (mb_input | mb_size_ctrl)
+     * is set to 1. The data in this buffer correspond to the input source. 16x16 MB is in raster scan order,
+     * each MB control data structure is defined by VAEncFEIMBControlH264.
+     * Buffer size shall not be less than the number of 16x16 blocks multiplied by
+     * sizeof(VAEncFEIMBControlH264).
+     * Note: if mb_qp is set, VAEncQpBufferH264 is expected.
```

Comment:
VAEncQpBufferH264 is per MB QP input buffer layout, it's for both legacy and FEI encoder, you can find it in va_enc_h264.h file in Carl's patch.